SYCL build failures for PV, intel/oneapi/2023.1.0: error: cannot convert 'const ...floating_point_wrapper<sycl::ext::oneapi::bfloat16>>' to 'float' without a conversion operator

**Describe the bug**
The nightly SYCL build with intel/oneapi/2023.1.0 is failing to compile due to conversion issues between bfloat half types and float with output:
```
01:30:11 /projects/x86-64-icelake-rocky8/compilers/intel-oneapi-compilers/2023.1.0/gcc/8.5.0/base/6g2jkiv/compiler/2023.1.0/linux/bin-llvm/../include/sycl/ext/oneapi/bfloat16.hpp:198:3: error: cannot convert 'const Kokkos::Experimental::Impl::BitComparisonWrapper<Kokkos::Experimental::Impl::floating_point_wrapper<sycl::ext::oneapi::bfloat16>>' to 'float' without a conversion operator
01:30:11   OP(bfloat16, /)
01:30:11   ^~~~~~~~~~~~~~~
01:30:11 /projects/x86-64-icelake-rocky8/compilers/intel-oneapi-compilers/2023.1.0/gcc/8.5.0/base/6g2jkiv/compiler/2023.1.0/linux/bin-llvm/../include/sycl/ext/oneapi/bfloat16.hpp:193:17: note: expanded from macro 'OP'
01:30:11     return type{static_cast<float>(lhs) op static_cast<float>(rhs)};           \
01:30:11                 ^~~~~~~~~~~~~~~~~~~~~~~
01:30:11 /home/jenkins/blake-new/workspace/Kokkos_develop_blake_sycl_pv_oneapi_2023_1_0/kokkos/core/unit_test/TestNumericTraits.hpp:118:27: note: in instantiation of function template specialization 'sycl::ext::oneapi::operator/<Kokkos::Experimental::Impl::BitComparisonWrapper<Kokkos::Experimental::Impl::floating_point_wrapper<sycl::ext::oneapi::bfloat16>>>' requested here
01:30:11     compare() = one + eps / 2;
01:30:11                           ^
01:30:11 /home/jenkins/blake-new/workspace/Kokkos_develop_blake_sycl_pv_oneapi_2023_1_0/kokkos/core/src/SYCL/Kokkos_SYCL_ParallelReduce_Range.hpp:111:15: note: in instantiation of member function 'TestNumericTraits<Kokkos::Experimental::SYCL, Kokkos::Experimental::Impl::floating_point_wrapper<sycl::ext::oneapi::bfloat16>, Epsilon>::operator()' requested here
01:30:11               functor(WorkTag(), begin, update);
...
```

This likely follows from #6590 , @dalg24 or @masterleinad can you give guidance?

**Please also include the following items to support reproducing the bug**
Reproducer (blake PV queue):
```bash
salloc -N 1 -p PV

module load cmake intel-oneapi-compilers/2023.1.0 intel-oneapi-dpl/2022.1.0 git

$KOKKOS_PATH/generate_makefile.bash --with-sycl --arch=INTEL_PVC --compiler=icpx --cxxflags="-fp-model=precise" --cmake-flags=-DKokkos_ENABLE_ONEDPL=OFF

make -j
```

--- a/core/unit_test/TestNumericTraits.hpp
+++ b/core/unit_test/TestNumericTraits.hpp
@@ -110,8 +110,8 @@ struct TestNumericTraits {
 
   KOKKOS_FUNCTION void operator()(Epsilon, int, int& e) const {
     using Kokkos::Experimental::epsilon;
-    auto const eps = epsilon<T>::value;
-    auto const one = T(1);
+    T const eps = epsilon<T>::value;
+    T const one = 1;
     // Avoid higher precision intermediate representation
     compare() = one + eps;
     e += (int)!(compare() != one);

